x-pack/plugins/uptime/server/lib/requests/search/refine_potential_matches.ts

```diff
@@ -116,13 +115,39 @@ export const mostRecentCheckGroups = async (
             location: {
               terms: { field: 'observer.geo.name', missing: 'N/A', size: 100 },
               aggs: {
-                top: {
-                  top_hits: {
-                    sort: [{ '@timestamp': 'desc' }],
-                    _source: {
-                      includes: ['monitor.check_group', '@timestamp', 'summary.up', 'summary.down'],
+                summaries: {
+                  // only match summary docs because we only want the latest *complete* check group.
+                  filter: { exists: { field: 'summary' } },
```

Comment:
i wonder why this filter is moved inside aggs? this will make terms aggs for observer.geo.name inefficient.

Reply:
This is necessary because we don't want to only match summaries for `latest_matching`. As far as efficiency goes, I think we're still going to be fine, because the cardinality of locations is low, and the backing store for terms is columnar.